ape_alchemy/provider.py

```diff
@@ -179,26 +220,60 @@ def get_virtual_machine_error(self, exception: Exception, **kwargs) -> VirtualMa
 
         return VirtualMachineError(message=message, txn=txn)
 
-    def _make_request(self, endpoint: str, parameters: Optional[List] = None) -> Any:
-        try:
-            return super()._make_request(endpoint, parameters)
-        except HTTPError as err:
-            response_data = err.response.json() if err.response else {}
-            if "error" not in response_data:
-                raise AlchemyProviderError(str(err)) from err
-
-            error_data = response_data["error"]
-            message = (
-                error_data.get("message", str(error_data))
-                if isinstance(error_data, dict)
-                else error_data
-            )
-            cls = (
-                AlchemyFeatureNotAvailable
-                if "is not available" in message
-                else AlchemyProviderError
-            )
-            raise cls(message) from err
+    def _make_request(
+        self,
+        endpoint: str,
+        parameters: Optional[List] = None,
+        min_retry_delay: Optional[int] = None,
+        retry_backoff_factor: Optional[int] = None,
+        max_retry_delay: Optional[int] = None,
+        max_retries: Optional[int] = None,
+        retry_jitter: Optional[int] = None,
+    ) -> Any:
+        alchemy_config = cast(AlchemyConfig, self.config_manager.get_config("alchemy"))
+        min_retry_delay = (
+            min_retry_delay if min_retry_delay is not None else alchemy_config.min_retry_delay
+        )
+        retry_backoff_factor = (
+            retry_backoff_factor
+            if retry_backoff_factor is not None
+            else alchemy_config.retry_backoff_factor
+        )
+        max_retry_delay = (
+            max_retry_delay if max_retry_delay is not None else alchemy_config.max_retry_delay
+        )
+        max_retries = max_retries if max_retries is not None else alchemy_config.max_retries
+        retry_jitter = retry_jitter if retry_jitter is not None else alchemy_config.retry_jitter
+        for attempt in range(max_retries):
+            try:
+                return super()._make_request(endpoint, parameters)
+            except HTTPError as err:
+                message = str(err)
+                if any(
+                    error in message
+                    for error in ["exceeded its compute units", "Too Many Requests for url"]
```

Comment:
I think this is the main part that changes per-provider (the specific error string to look for). I am thinking we may do what you suggested and move some of this to core when using default node connections (ape-geth), we can still receive this benefit.

Maybe we can a method to the base class that can be overriden like:

```python
def is_rate_limit_error(self, err):
```